### PR TITLE
[TIMOB-4388] Android: Pressure(Force)/Size for Views

### PR DESCRIPTION
**Pull request by m1ga**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-4388

Adding getPressure()/getSize() in the touchevents.

### CODE REVIEW

**Comment by m1ga** [comment]
I've accidentally added the other PR again so I've removed it here, that's why TUIText.java is shown as 'changed' again. Hope that's not a problem

File: apidoc/Titanium/UI/View.yml [hunk 1]
```diff
@@ -404,9 +404,15 @@ events:
       - name: force
         summary: |
             The current force value of the touch event.
-            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later.
+            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later and on some android devices
```
Comment (hansemannn) [inline apidoc/Titanium/UI/View.yml]: "some android devices" sounds a bit odd, can you describe that a bit better? Also please capitalize the "Android" and set the period (`.`) at the end of the sentence.

File: apidoc/Titanium/UI/View.yml [hunk 2]
```diff
@@ -404,9 +404,15 @@ events:
...
         type: Number
-
+        
+      - name: size
+        summary: |
+            The current size of the touch area.
+            Note: This property is only available android devices
```
Comment (hansemannn) [inline apidoc/Titanium/UI/View.yml]: Missing  period after "devices". Also we can put the "Note" in the same line.

File: apidoc/Titanium/UI/View.yml [hunk 3]
```diff
@@ -443,7 +449,13 @@ events:
...
         summary: |
             The current force value of the touch event.
-            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later.
+            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later and on some android devices
+        type: Number
+        
+      - name: size
+        summary: |
+            The current size of the touch area.
+            Note: This property is only available android devices
```
Comment (hansemannn) [inline apidoc/Titanium/UI/View.yml]: Same as above

File: apidoc/Titanium/UI/View.yml [hunk 4]
```diff
@@ -481,9 +493,15 @@ events:
...
             The current force value of the touch event.
-            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later.
+            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later and on some android devices
         type: Number
-
+        
+      - name: size
+        summary: |
+            The current size of the touch area.
+            Note: This property is only available android devices
```
Comment (hansemannn) [inline apidoc/Titanium/UI/View.yml]: Same as above

File: apidoc/Titanium/UI/View.yml [hunk 5]
```diff
@@ -517,7 +535,13 @@ events:
...
         summary: |
             The current force value of the touch event.
-            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later.
+            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later and on some android devices
+        type: Number
+        
+      - name: size
+        summary: |
+            The current size of the touch area.
+            Note: This property is only available android devices
```
Comment (hansemannn) [inline apidoc/Titanium/UI/View.yml]: Sams as above

File: apidoc/Titanium/UI/View.yml (hunk 1 above)
Comment (m1ga) [inline apidoc/Titanium/UI/View.yml]: removed the "some". The Android documentation says you'll get a value. So it should just be 0 on devices that don't feature pressure

File: apidoc/Titanium/UI/View.yml (hunk 2 above)
Comment (m1ga) [inline apidoc/Titanium/UI/View.yml]: Sure to put it in the same line? All other "Note"'s are always in the next line

Comment (hansemannn) [inline apidoc/Titanium/UI/View.yml]: The rule to follow there is, that the text per line doesn't extend a certain length (I mostly just compare it to existing lines). It doesn't change much, just a thing that I'm personally used to

Comment (m1ga) [inline apidoc/Titanium/UI/View.yml]: I see. No problem, changed it!

**Comment by ashcoding** [comment]
@m1ga Just curious, what versions of Android and models of device have you tested this with?

**Comment by m1ga** [comment]
**HTC A9 (6.0.1):** Force and size are different (force looks like the actual size of the touched area)
**Motorola Moto G (2nd gen, 5.0.2):** Force and size are the same value (size of the touched area)
**Samsung Galaxy S7:** Force always 1 and size value lower compared to the other size values but still shows the size of the touched area

File: apidoc/Titanium/UI/View.yml [hunk 6]
```diff
@@ -443,7 +448,12 @@ events:
       - name: force
         summary: |
             The current force value of the touch event.
-            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later.
+            Note: This property is only available for iOS devices that support 3D-Touch and run 9.0 or later and on Android devices.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/View.yml]: Please leave a note in all docs here, that it's only available on **some** Android devices.

File: apidoc/Titanium/UI/View.yml [hunk 7]
```diff
@@ -443,7 +448,12 @@ events:
...
+        type: Number
+        
+      - name: size
+        summary: |
+            The current size of the touch area. Note: This property is only available Android devices.
```
Comment (hansemannn) [inline apidoc/Titanium/UI/View.yml]: Typo, missing "on" in "This property is only available Android devices.", also same missing note about **some** devices.

**Comment by hansemannn** [comment]
Left some docs comments, also please pull latest master to resolve merge conflicts.

**Comment by m1ga** [comment]
:sweat: merged the master branch...now it should compile again and I didn't break anything :smile:

**Comment by hansemannn** [comment]
Approved!

**Pull request merged**